InSimDotNet/PacketReader.cs

```diff
@@ -103,6 +112,34 @@ public class PacketReader {
             return BitConverter.ToSingle(buffer, position - 4);
         }
 
+        /// <summary>
+        /// Reads a LFS Car name or modded car skinID (as of 0.6W)
+        /// </summary>
+        /// <param name="count">The number of bytes to read.</param>
+        /// <returns>A Unicode string.</returns>
+        public string ReadCNameString(int count = 4)
+        {
+            position += count;
+
+            var buf = new byte[4];
+            Array.Copy(buffer, position - count, buf, 0, count);
+
+            if (IsAlphaNumeric(buf[0]) && IsAlphaNumeric(buf[1]) && IsAlphaNumeric(buf[2]))
+            {
+                return LfsEncoding.Current.GetString(buffer, position - count, count);
+            }
+
+            return buf[2].ToString("X2") + buf[1].ToString("X2") + buf[0].ToString("X2");
+
+            bool IsAlphaNumeric(byte b)

```

Comment:
Also there is a built-in method something like Char.IsLetterOrNumber() that could probably replace this.

Reply:
Yes, probably - I just took the sample c++ code from scawen.
Although it should be tested before using, since some 127+ ascii code characters could be interpreted as valid letters.
`À` (ascii 0xC0 in latin1 codepage) for example.